interface/repository/pkmn_repository_test.go

```diff
@@ -0,0 +1,33 @@
+package repository
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/jvillarreal-w/academy-go-q42021/common"
+	"github.com/jvillarreal-w/academy-go-q42021/domain/model"
+	u "github.com/jvillarreal-w/academy-go-q42021/utils"
+)
+
+func TestPokemonRepository_FindAll(t *testing.T) {
+	pr := NewPokemonRepository(common.TestDataSourcePath)
+	rows, _ := readInternalDataSource(common.TestDataSourcePath)
+	var p []*model.Pokemon
+
+	if p, _ = pr.FindAll(p); len(p) != len(rows) {
+		t.Errorf("Expected: %v but got: %v", len(rows), len(p))
+	}
+}
+
+func TestPokemonRepository_FindById(t *testing.T) {
```

Comment:
For testing you would like to have multiple cases to test every single(or at least most of) possible thing that can happen in your functions, maybe if you pass an invalid value the function would return an error, if you pass a value with some different values you would receive different stuff and so on.

The way to do this is that we have like an array of test cases, where every element  defines the arguments that your function takes, and things that can change per test. and then we iterate every test and run then.

Maybe this article could help https://www.myhatchpad.com/insight/go-unit-testing-at-compass/ or if needed we could schedule a mentoring session to look at how to do testing more effectively